pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java

```diff
@@ -79,6 +83,14 @@ void addBadge(Badge badge) {
         return Collections.unmodifiableSet(badges);
     }
 
+    public Set<Pokemon> getPokemons() {
+        return Collections.unmodifiableSet(pokemons);
+    }
+
+    public void setPokemons(Set<Pokemon> pokemons) {
```

Comment:
You don't want to set whole collection to replace the current one. Just do `addPokemon` method please which will add a new Pokemon to currect Collection. :)